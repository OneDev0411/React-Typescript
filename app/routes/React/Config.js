// Config.js
import React from 'react'
import { Route, IndexRoute, Redirect } from 'react-router'

// Main component
import App from '../../components/App'

// Pages
import Landing from '../../components/Pages/Landing'
import Recents from '../../components/Pages/Dashboard/Recents'
import Mls from '../../components/Pages/Dashboard/Mls'
import Contacts from '../../components/Pages/Dashboard/Contacts'
import Tasks from '../../components/Pages/Dashboard/Tasks'
import Transactions from '../../components/Pages/Dashboard/Transactions'
import SingleTransaction from '../../components/Pages/Dashboard/Transactions/Single'
import NewTransaction from '../../components/Pages/Dashboard/Transactions/New'
import SignUp from '../../components/Pages/SignUp'
import SignIn from '../../components/Pages/SignIn'
import Verify from '../../components/Pages/Verify'
import Password from '../../components/Pages/Password'
import NoMatch from '../../components/Pages/NoMatch'
import Settings from '../../components/Pages/Account/Settings'
import Notifications from '../../components/Pages/Account/Notifications'
import Recommend from '../../components/Pages/Dashboard/Mls/Listing/Recommend'
import Agents from '../../components/Pages/Dashboard/Mls/Agents'

export default (
  <Route path="/" component={App}>
    <IndexRoute component={Landing}/>
    <Route path="signup" component={SignUp}/>
    <Route path="signin" component={SignIn}/>
    <Route path="account/settings" component={Settings}/>
    <Route path="account/notifications" component={Notifications}/>
    <Redirect from="dashboard" to="dashboard/recents" />
    <Route path="dashboard/recents" component={Recents}/>
    <Route path="dashboard/recents/:room_id" component={Recents}/>
    <Route path="dashboard/mls" component={Mls}/>
    <Route path="dashboard/mls/agents" component={Agents}/>
    <Route path="dashboard/mls/listing/recommend" component={Recommend}/>
    <Route path="dashboard/contacts" component={Contacts}/>
    <Route path="dashboard/contacts/:id" component={Contacts}/>
    <Route path="dashboard/tasks" component={Tasks}/>
    <Route path="dashboard/transactions" component={Transactions}/>
    <Route path="dashboard/transactions/new" component={NewTransaction}/>
    <Route path="dashboard/transactions/:id" component={SingleTransaction}/>
    <Route path="dashboard/transactions/:id/attachments/:id" component={SingleTransaction}/>
    <Route path="verify/:slug" component={Verify}/>
    <Route path="password/:slug" component={Password}/>
    <Route path="*" component={NoMatch}/>
  </Route>
)