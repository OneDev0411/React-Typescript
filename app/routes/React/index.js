// index.js
import React from 'react'
import { match, RoutingContext } from 'react-router'
import ReactDOMServer from 'react-dom/server'

// Store
import AppStore from '../../stores/AppStore'

// Config
import routes from './Config'

module.exports = (app, config) => {

  app.get('*',(req, res) => {

    match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
      
      let reactMarkup = ReactDOMServer.renderToStaticMarkup(<RoutingContext data={AppStore.data} {...renderProps} />)
      
      res.locals.reactMarkup = reactMarkup

      console.log(reactMarkup)

      if (error) {
      
        res.status(500).send(error.message)
      
      } else if (redirectLocation) {
        
        res.redirect(302, redirectLocation.pathname + redirectLocation.search)

      } else if (renderProps) {
        
        // Success!
        res.status(200).render('index.html')
      
      } else {

        res.status(404).render('index.html')
      
      }

    })
  })
  
}