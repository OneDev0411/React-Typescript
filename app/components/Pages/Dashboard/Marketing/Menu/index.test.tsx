import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Menu } from './index'

describe('MC Menu', () => {
  it('renders list of templates type', () => {
    expect(toJson(shallow(
      <Menu />
    ))).toMatchSnapshot()
  })
})
