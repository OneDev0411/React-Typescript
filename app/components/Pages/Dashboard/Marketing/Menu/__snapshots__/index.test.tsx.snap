// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MC Menu renders list of templates type 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin-bottom: 2.5em;"
      >
        <a
          class="onboarding--all-designs ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          <svg
            class="IconAllDesigns__Image-wgzmtq-0 cypocS"
            style="margin-right: 0.5rem;"
            viewBox="0 0 16 16"
          >
            <g
              fill-rule="evenodd"
            >
              <path
                d="M9.333 4.333A.667.667 0 0 0 10 3.667V1.333C10 .597 9.403 0 8.667 0H1.333C.597 0 0 .597 0 1.333v10.334C0 12.403.597 13 1.333 13h.334a.667.667 0 1 0 0-1.333.333.333 0 0 1-.334-.334V1.667c0-.184.15-.334.334-.334h6.666c.184 0 .334.15.334.334v2c0 .368.298.666.666.666z"
              />
              <path
                d="M3.333 2.833a.5.5 0 1 0-1 0v6a.5.5 0 1 0 1 0v-6z"
              />
              <circle
                cx="5"
                cy="3.333"
                r="1"
              />
              <circle
                cx="10.667"
                cy="8.333"
                r="1"
              />
              <path
                d="M14.667 5.333h-8c-.737 0-1.334.597-1.334 1.334v1.666a.667.667 0 0 0 1.334 0V7c0-.184.149-.333.333-.333h7.333c.184 0 .334.149.334.333v4c0 .184-.15.333-.334.333H13a.667.667 0 0 0 0 1.334h1.667c.736 0 1.333-.597 1.333-1.334V6.667c0-.737-.597-1.334-1.333-1.334z"
              />
              <path
                d="M3.333 11.333v3.334c0 .736.597 1.333 1.334 1.333H10c.736 0 1.333-.597 1.333-1.333v-3.334c0-.736-.597-1.333-1.333-1.333H4.667c-.737 0-1.334.597-1.334 1.333zm6.667 3c0 .184-.15.334-.333.334H5a.333.333 0 0 1-.333-.334v-1.282a.167.167 0 0 1 .228-.154l2.253.903c.12.047.252.047.37 0l2.253-.901a.167.167 0 0 1 .229.154v1.28zm-5-3h4.667c.184 0 .333.15.333.334 0 .037-.023.07-.057.084l-2.548 1.02a.173.173 0 0 1-.124 0l-2.547-1.02a.091.091 0 0 1-.057-.084c0-.184.149-.334.333-.334z"
                fill-rule="nonzero"
              />
            </g>
          </svg>
          <div
            class="styled__ListItemName-q32m1c-4 iCFSOs"
          >
            All Designs
          </div>
        </a>
        <a
          class="onboarding--my-designs ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          <svg
            class="IconMyDesigns__Image-sc-16mri3k-0 kLTwEN"
            style="margin-right: 0.5rem;"
            viewBox="0 0 16 16"
          >
            <g
              fill-rule="evenodd"
            >
              <path
                d="M10.88 12.16a1.28 1.28 0 0 0 1.28-1.28v-9.6A1.28 1.28 0 0 0 10.88 0h-9.6A1.28 1.28 0 0 0 0 1.28v9.6c0 .707.573 1.28 1.28 1.28h9.6zM1.28 1.6a.32.32 0 0 1 .32-.32h8.96a.32.32 0 0 1 .32.32v8.96a.32.32 0 0 1-.32.32H1.6a.32.32 0 0 1-.32-.32V1.6z"
                fill-rule="nonzero"
              />
              <path
                d="M6.38 3.251a.218.218 0 0 0-.28 0L3.277 5.704a.21.21 0 0 0-.064.236c.032.085.114.14.204.14h.423a.32.32 0 0 1 .32.32v1.92c0 .176.144.32.32.32h.64a.32.32 0 0 0 .32-.32v-.8a.8.8 0 1 1 1.6 0v.8c0 .176.144.32.32.32H8a.32.32 0 0 0 .32-.32V6.4a.32.32 0 0 1 .32-.32h.423c.09 0 .172-.056.204-.14a.21.21 0 0 0-.064-.237L6.38 3.251zm8.975 2.659a1.28 1.28 0 0 0-1.16-1.39l-.64-.058a.64.64 0 1 0-.115 1.275l.32.028a.32.32 0 0 1 .29.348l-.698 7.647a.32.32 0 0 1-.347.29L4.4 13.265a.64.64 0 1 0-.113 1.274l8.922.816a1.28 1.28 0 0 0 1.392-1.16l.753-8.286z"
              />
            </g>
          </svg>
          <div
            class="styled__ListItemName-q32m1c-4 iCFSOs"
          >
            My Designs
          </div>
        </a>
      </div>
      <div
        class="onboarding--menus"
        data-test="mc-store-menu"
      >
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          As Seen In
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Birthday
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Brand Campaigns
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Coming Soon
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Holiday
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Just Listed
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Under Contract
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Just Sold
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Multi Listings
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          New Agent
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          New Price
        </a>
        <a
          class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
        >
          Open House
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin-bottom: 2.5em;"
    >
      <a
        class="onboarding--all-designs ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        <svg
          class="IconAllDesigns__Image-wgzmtq-0 cypocS"
          style="margin-right: 0.5rem;"
          viewBox="0 0 16 16"
        >
          <g
            fill-rule="evenodd"
          >
            <path
              d="M9.333 4.333A.667.667 0 0 0 10 3.667V1.333C10 .597 9.403 0 8.667 0H1.333C.597 0 0 .597 0 1.333v10.334C0 12.403.597 13 1.333 13h.334a.667.667 0 1 0 0-1.333.333.333 0 0 1-.334-.334V1.667c0-.184.15-.334.334-.334h6.666c.184 0 .334.15.334.334v2c0 .368.298.666.666.666z"
            />
            <path
              d="M3.333 2.833a.5.5 0 1 0-1 0v6a.5.5 0 1 0 1 0v-6z"
            />
            <circle
              cx="5"
              cy="3.333"
              r="1"
            />
            <circle
              cx="10.667"
              cy="8.333"
              r="1"
            />
            <path
              d="M14.667 5.333h-8c-.737 0-1.334.597-1.334 1.334v1.666a.667.667 0 0 0 1.334 0V7c0-.184.149-.333.333-.333h7.333c.184 0 .334.149.334.333v4c0 .184-.15.333-.334.333H13a.667.667 0 0 0 0 1.334h1.667c.736 0 1.333-.597 1.333-1.334V6.667c0-.737-.597-1.334-1.333-1.334z"
            />
            <path
              d="M3.333 11.333v3.334c0 .736.597 1.333 1.334 1.333H10c.736 0 1.333-.597 1.333-1.333v-3.334c0-.736-.597-1.333-1.333-1.333H4.667c-.737 0-1.334.597-1.334 1.333zm6.667 3c0 .184-.15.334-.333.334H5a.333.333 0 0 1-.333-.334v-1.282a.167.167 0 0 1 .228-.154l2.253.903c.12.047.252.047.37 0l2.253-.901a.167.167 0 0 1 .229.154v1.28zm-5-3h4.667c.184 0 .333.15.333.334 0 .037-.023.07-.057.084l-2.548 1.02a.173.173 0 0 1-.124 0l-2.547-1.02a.091.091 0 0 1-.057-.084c0-.184.149-.334.333-.334z"
              fill-rule="nonzero"
            />
          </g>
        </svg>
        <div
          class="styled__ListItemName-q32m1c-4 iCFSOs"
        >
          All Designs
        </div>
      </a>
      <a
        class="onboarding--my-designs ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        <svg
          class="IconMyDesigns__Image-sc-16mri3k-0 kLTwEN"
          style="margin-right: 0.5rem;"
          viewBox="0 0 16 16"
        >
          <g
            fill-rule="evenodd"
          >
            <path
              d="M10.88 12.16a1.28 1.28 0 0 0 1.28-1.28v-9.6A1.28 1.28 0 0 0 10.88 0h-9.6A1.28 1.28 0 0 0 0 1.28v9.6c0 .707.573 1.28 1.28 1.28h9.6zM1.28 1.6a.32.32 0 0 1 .32-.32h8.96a.32.32 0 0 1 .32.32v8.96a.32.32 0 0 1-.32.32H1.6a.32.32 0 0 1-.32-.32V1.6z"
              fill-rule="nonzero"
            />
            <path
              d="M6.38 3.251a.218.218 0 0 0-.28 0L3.277 5.704a.21.21 0 0 0-.064.236c.032.085.114.14.204.14h.423a.32.32 0 0 1 .32.32v1.92c0 .176.144.32.32.32h.64a.32.32 0 0 0 .32-.32v-.8a.8.8 0 1 1 1.6 0v.8c0 .176.144.32.32.32H8a.32.32 0 0 0 .32-.32V6.4a.32.32 0 0 1 .32-.32h.423c.09 0 .172-.056.204-.14a.21.21 0 0 0-.064-.237L6.38 3.251zm8.975 2.659a1.28 1.28 0 0 0-1.16-1.39l-.64-.058a.64.64 0 1 0-.115 1.275l.32.028a.32.32 0 0 1 .29.348l-.698 7.647a.32.32 0 0 1-.347.29L4.4 13.265a.64.64 0 1 0-.113 1.274l8.922.816a1.28 1.28 0 0 0 1.392-1.16l.753-8.286z"
            />
          </g>
        </svg>
        <div
          class="styled__ListItemName-q32m1c-4 iCFSOs"
        >
          My Designs
        </div>
      </a>
    </div>
    <div
      class="onboarding--menus"
      data-test="mc-store-menu"
    >
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        As Seen In
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Birthday
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Brand Campaigns
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Coming Soon
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Holiday
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Just Listed
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Under Contract
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Just Sold
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Multi Listings
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        New Agent
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        New Price
      </a>
      <a
        class="ActiveItem__Item-sc-1ihy29z-0 deglPM"
      >
        Open House
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
